shared/src/sources/balancer/info_fetching.rs

```diff
@@ -109,13 +110,18 @@ impl PoolInfoFetching for PoolInfoFetcher {
 
         let tokens = token_data.await?.0;
         let scaling_exponents = self.get_scaling_exponents(&tokens).await?;
+        let amplification_data = amplification_parameter.await?;
```

Comment:
Note that you can also destructure the tuple here:
```suggestion
        let (amplification_factor, _, precision) = amplification_parameter.await?;
```

Not a suggestion for change, (I'm fine how it is).

Reply:
Actually I like this better and will include it before merge.